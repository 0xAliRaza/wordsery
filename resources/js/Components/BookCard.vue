<template>
    <div class="card">
        <img
            v-if="book.thumbnail_link"
            :src="book.thumbnail_link"
            :alt="book.title"
            :class="{ thumb: true, 'thumb--compact': compact }"
        />
        <div class="meta">
            <div class="title">
                {{ book.title }}
            </div>
            <div class="authors">
                <template
                    v-for="(author, index) in book.authors"
                    v-bind:key="index"
                >
                    <template v-if="index > 0">, </template>{{ author.name }}
                </template>
            </div>
        </div>
    </div>
</template>

<script>
export default {
    props: {
        book: {
            type: Object,
            required: true,
        },
        compact: Boolean,
    },
    setup() {},
};
</script>

<style lang="scss" scoped>
@import "@scss/abstracts";
.card {
    background-color: $light;
    display: flex;
    border: 1px solid $gray-3;
    border-radius: 2px;
    overflow: hidden;
}
.thumb {
    width: 100%;
    max-width: pxToRem(60);
    height: auto;
    &--compact {
        max-width: pxToRem(50);
    }
}
.meta {
    display: flex;
    flex: 1;
    flex-direction: column;
    justify-content: center;
    padding: pxToRem(16);
}
.title {
    font-size: pxToRem(16);
    color: $primary;
    font-weight: 600;
    margin-bottom: 3px;
}
.authors {
    font-size: pxToRem(14);
    color: $gray-6;
}
</style>
