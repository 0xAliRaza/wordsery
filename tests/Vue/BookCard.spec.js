import { shallowMount } from "@vue/test-utils";
import BookCard from "@/Components/BookCard";
import book from "./book.json";


const factory = async (props = { book }) => {
    return await shallowMount(BookCard, {
        props: { ...props },
    });
};

test("renders correctly", () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
});

test("contains title", () => {
    const wrapper = factory();
    expect(wrapper.html()).toContain(book.volumeInfo.title);
});

test("contains authors", () => {
    const wrapper = factory();
    for (let i = 0; i < book.volumeInfo.authors.length; i++) {
        const author = book.volumeInfo.authors[i];
        expect(wrapper.html()).toContain(author);
    }
});

test("renders thumbnail", () => {
    const wrapper = factory();
    expect(wrapper.find("img").exists());
});

const alteredBook = book;
alteredBook.volumeInfo.imageLinks.smallThumbnail = undefined;
test("does not render thumbnail", () => {
    const wrapper = factory({ book: alteredBook });
    expect(wrapper.find("img").exists()).toBeFalsy();
});
