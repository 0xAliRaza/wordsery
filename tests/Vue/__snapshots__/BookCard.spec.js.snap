// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div class="card">
  <!--v-if-->
  <div class="meta">
    <div class="title"></div>
    <div class="authors"></div>
  </div>
</div>
`;
